examples/chat_with_document/main.go

```diff
@@ -0,0 +1,154 @@
+/*
+Copyright 2023 The KubeAGI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"github.com/gofiber/fiber/v2"
+	"github.com/gofiber/fiber/v2/middleware/cors"
+	"github.com/tmc/langchaingo/schema"
+
+	zhipuaiembeddings "github.com/kubeagi/arcadia/pkg/embeddings/zhipuai"
+	"github.com/kubeagi/arcadia/pkg/llms/zhipuai"
+	"github.com/kubeagi/arcadia/pkg/vectorstores/chromadb"
+)
+
+var (
+	apiKey = flag.String("api-key", "", "used to connect to ZhiPuAI platform")
+	url    = flag.String("url", "", "the chromaDB scheme")
+	addr   = flag.String("addr", ":8800", "used to listen and serve GET request, default 8800")
+)
+
+func main() {
+	flag.Parse()
+
+	if err := run(); err != nil {
+		fmt.Printf("Run failed, error:\n %v", err)
+	}
+}
+
+func run() error {
+	fmt.Println("Starting chat server example...")
+
+	// check ZhiPuAI api key, build embedder
+	if *apiKey == "" {
+		return fmt.Errorf("ZhiPuAI api key is empty")
+	}
+
+	fmt.Println("Connecting platform...")
+	z := zhipuai.NewZhiPuAI(*apiKey)
+	_, err := z.Validate()
+	if err != nil {
+		return fmt.Errorf("error validating ZhiPuAI api key: %s", err.Error())
+	}
+
+	embedder, err := zhipuaiembeddings.NewZhiPuAI(zhipuaiembeddings.WithClient(*z))
+	if err != nil {
+		return fmt.Errorf("error creating embedder: %s", err.Error())
+	}
+
+	// check scheme & host, connect chromaDB server
+	if *url == "" {
+		return fmt.Errorf("chromaDB scheme is empty")
+	}
+
+	fmt.Println("Connecting vector database...")
+	db, err := chromadb.New(
+		chromadb.WithURL(*url),
+		chromadb.WithEmbedder(embedder),
+		chromadb.WithNameSpace("arcadia"),
+	)
+	if err != nil {
+		return fmt.Errorf("error creating chroma db: %s", err.Error())
+	}
+
+	fmt.Println("Creating HTTP server...")
+	app := fiber.New(fiber.Config{
+		AppName:       "chat-server",
+		CaseSensitive: true,
+		StrictRouting: true,
+		Immutable:     true,
+	})
+
+	app.Use(cors.New(cors.ConfigDefault))
+
+	app.Get("/", HomePageGetHandler)
+	app.Post("/load", LoadHandler)
```

Comment:
same as `/chat` we use our pre-confiugred apikey and chromadb for loading documents